analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/RestrictDeserializedTypes.cs

```diff
@@ -26,5 +26,20 @@ public sealed class RestrictDeserializedTypes : RestrictDeserializedTypesBase
 
     protected override DiagnosticDescriptor Rule => S5773;
 
-    public override bool ShouldExecute() => false;
+    public override bool ShouldExecute() => true;
+
+    protected override bool IsBindToTypeMethod(SyntaxNode methodDeclaration) =>
```

Comment:
**Optional**: this can be simplified by just checking if the method has the override keyword. It will also improve the performance, as we no longer have to request the semantic model every time. 
**Downside**: the highly unlikely scenario that someone will add an intermediate class derived from `SerializationBinder` with an abstract `IsBindToType` method with different parameters then derive the binder from that class and override that method in it.
I think it's a good trade-off. WDYT?

```cs
protected override bool IsBindToTypeMethod(SyntaxNode methodDeclaration) =>
    methodDeclaration is MethodDeclarationSyntax { Identifier.Text: "BindToType", ParameterList.Parameters.Count: 2 } syntax
    && syntax.Modifiers.Any(x => x.IsKind(SyntaxKind.OverrideKeyword));
```

Reply:
The biggest downside is that it makes the code's intentions extremely unclear. Instead of doing premature optimization here, I'll keep it the way it is.

Reply:
`BindToName`  is virtual, so an implementation could also have `override`.